pubsub/kubemq/kubemq_events.go

```diff
@@ -0,0 +1,169 @@
+package kubemq
+
+import (
+	"context"
+	"github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/kit/logger"
+	"github.com/kubemq-io/kubemq-go"
+	"github.com/kubemq-io/kubemq-go/pkg/uuid"
+	"time"
+)
+
+type kubemqEventsClient interface {
+	Stream(ctx context.Context, onError func(err error)) (func(msg *kubemq.Event) error, error)
+	Subscribe(ctx context.Context, request *kubemq.EventsSubscription, onEvent func(msg *kubemq.Event, err error)) error
+	Close() error
+}
+
+type kubeMQEvents struct {
+	client               kubemqEventsClient
+	metadata             *metadata
+	logger               logger.Logger
+	publishFunc          func(event *kubemq.Event) error
+	resultChan           chan error
+	waitForResultTimeout time.Duration
+	ctx                  context.Context
+	ctxCancel            context.CancelFunc
+	isInitialized        bool
+}
+
+func newkubeMQEvents(logger logger.Logger) *kubeMQEvents {
+	return &kubeMQEvents{
+		client:               nil,
+		metadata:             nil,
+		logger:               logger,
+		publishFunc:          nil,
+		resultChan:           make(chan error, 1),
+		waitForResultTimeout: 60 * time.Second,
+		ctx:                  nil,
+		ctxCancel:            nil,
+	}
+}
+func (k *kubeMQEvents) init() error {
+	k.ctx, k.ctxCancel = context.WithCancel(context.Background())
+	if k.metadata.useMock {
+		k.isInitialized = true
+		return nil
+	}
+	clientID := k.metadata.clientID
+	if clientID == "" {
+		clientID = uuid.New()
+	}
+	client, err := kubemq.NewEventsClient(k.ctx,
+		kubemq.WithAddress(k.metadata.host, k.metadata.port),
+		kubemq.WithClientId(clientID),
+		kubemq.WithTransportType(kubemq.TransportTypeGRPC),
+		kubemq.WithCheckConnection(true),
+		kubemq.WithAuthToken(k.metadata.authToken),
+		kubemq.WithAutoReconnect(true),
+		kubemq.WithReconnectInterval(time.Second))
+	if err != nil {
+		k.logger.Errorf("error init kubemq client error: %s", err.Error())
+		return err
+	}
+	k.ctx, k.ctxCancel = context.WithCancel(context.Background())
+	k.client = client
+	if err := k.setPublishStream(); err != nil {
+		k.logger.Errorf("error init kubemq client error: %w", err.Error())
+		return err
+	}
+	k.isInitialized = true
+	return nil
+}
+func (k *kubeMQEvents) Init(meta *metadata) error {
+	k.metadata = meta
+	_ = k.init()
```

Comment:
Why is the error discarded?

Reply:
This is because in events mode the SDK is never return error. this was done to be align with SDK signature

Reply:
Ok, as long as the error is logged